src/plugins/index_pattern_field_editor/public/components/preview/field_preview.tsx

```diff
@@ -5,78 +5,94 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
-import React from 'react';
-import { EuiTitle, EuiSpacer } from '@elastic/eui';
+import React, { useState, useCallback } from 'react';
+import { EuiSpacer, EuiResizeObserver } from '@elastic/eui';
+
+import { useFieldPreviewContext } from './field_preview_context';
+import { FieldPreviewHeader } from './field_preview_header';
+import { FieldPreviewEmptyPrompt } from './field_preview_empty_prompt';
+import { DocumentsNavPreview } from './documents_nav_preview';
+import { FieldPreviewError } from './field_preview_error';
+import { PreviewListItem } from './field_list/field_list_item';
+import { PreviewFieldList } from './field_list/field_list';
+
+import './field_preview.scss';
 
 export const FieldPreview = () => {
+  const [fieldListHeight, setFieldListHeight] = useState(-1);
+
+  const {
+    params: {
+      value: { name, script, format },
+    },
+    previewCount,
+    fields,
+    error,
+  } = useFieldPreviewContext();
+
+  // To show the preview we at least need a name to be defined, the script or the format
+  // and an first response from the _execute API
+  const isEmptyPromptVisible =
+    previewCount === 0
+      ? true
+      : error !== null || fields.length > 0
+      ? false
+      : name === null || (script === null && format === null);
+
+  const onFieldListResize = useCallback(({ height }: { height: number }) => {
+    setFieldListHeight(height);
+  }, []);
+
+  const renderPinnedFields = () => {
+    if (fields.length === 0) {
+      return null;
+    }
+
+    return (
+      <div>
+        <PreviewListItem
+          field={{ key: fields[0].key, value: JSON.stringify(fields[0].value) }}
+          highlighted
+        />
+      </div>
```

Comment:
Looks like this is rendering a list – can we make it a semantic list as well? (A `<ul>` with child `<li>` elements)

Reply:
Good point 👍